shell/prompt.go

```diff
@@ -3,12 +3,20 @@ package shell
 import (
 	"bufio"
 	"fmt"
-	"github.com/gruntwork-io/terragrunt/errors"
-	"github.com/gruntwork-io/terragrunt/options"
 	"os"
 	"strings"
+
+	"github.com/gruntwork-io/terragrunt/errors"
+	"github.com/gruntwork-io/terragrunt/options"
+	"golang.org/x/crypto/ssh/terminal"
 )
 
+// CanAcceptPrompt tells whether the program can receive input on the standard
+// input.
+func CanAcceptPrompt() bool {
+	return terminal.IsTerminal(int(os.Stdin.Fd()))
```

Comment:
Does this work on Windows?

Reply:
There is code that at least attempts to handle this from the Windows side... from the source code

```go
// IsTerminal returns true if the given file descriptor is a terminal.
func IsTerminal(fd int) bool {
	var st uint32
	err := windows.GetConsoleMode(windows.Handle(fd), &st)
	return err == nil
}
```

I have no idea whether that is the correct code, however, as I don't have a Windows machine.

It seems like the consequence of getting this wrong is not that big, though? If we incorrectly return true, we will wait to prompt when a Windows user can't enter any data. If we incorrectly return false, we will return an error, instead of prompting the user to enter a bucket.

Reply:
Fair enough!